cmd/openstack-populator/openstack-populator.go

```diff
@@ -3,66 +3,65 @@ package main
 import (
 	"flag"
 	"io"
-	"net/http"
 	"os"
 	"strings"
 	"time"
 
 	libclient "github.com/konveyor/forklift-controller/pkg/lib/client/openstack"
+	"github.com/konveyor/forklift-controller/pkg/metrics"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
 	"k8s.io/klog/v2"
 )
 
+type AppConfig struct {
+	identityEndpoint string
+	imageID          string
+	crNamespace      string
+	crName           string
+	secretName       string
+	ownerUID         string
+	pvcSize          int64
+	volumePath       string
+}
+
 func main() {
-	var (
-		identityEndpoint string
-		imageID          string
-		crNamespace      string
-		crName           string
-		secretName       string
-
-		volumePath string
-	)
+	config := &AppConfig{}
+	flag.StringVar(&config.identityEndpoint, "endpoint", "", "endpoint URL (https://openstack.example.com:5000/v2.0)")
+	flag.StringVar(&config.secretName, "secret-name", "", "secret containing OpenStack credentials")
+	flag.StringVar(&config.imageID, "image-id", "", "Openstack image ID")
+	flag.StringVar(&config.volumePath, "volume-path", "", "Path to populate")
+	flag.StringVar(&config.crName, "cr-name", "", "Custom Resource instance name")
+	flag.StringVar(&config.crNamespace, "cr-namespace", "", "Custom Resource instance namespace")
+	flag.StringVar(&config.ownerUID, "owner-uid", "", "Owner UID (usually PVC UID)")
+	flag.Int64Var(&config.pvcSize, "pvc-size", 0, "Size of pvc (in bytes)")
+	flag.Parse()
 
-	klog.InitFlags(nil)
+	if config.pvcSize <= 0 {
+		klog.Fatal("pvc-size must be greater than 0")
+	}
 
-	// Main arg
-	flag.StringVar(&identityEndpoint, "endpoint", "", "endpoint URL (https://openstack.example.com:5000/v2.0)")
-	flag.StringVar(&secretName, "secret-name", "", "secret containing OpenStack credentials")
+	certsDirectory, err := os.MkdirTemp("", "certsdir")
+	if err != nil {
+		klog.Fatal(err)
+	}
 
-	flag.StringVar(&imageID, "image-id", "", "Openstack image ID")
-	flag.StringVar(&volumePath, "volume-path", "", "Path to populate")
-	flag.StringVar(&crName, "cr-name", "", "Custom Resource instance name")
-	flag.StringVar(&crNamespace, "cr-namespace", "", "Custom Resource instance namespace")
+	defer os.RemoveAll(certsDirectory)
```

Comment:
wouldn't certdir anyway disappear when the container completes?

Reply:
yes but CDI does that
https://github.com/kubevirt/containerized-data-importer/blob/b9ae264f41dba6079cba32788be730004dd014fe/cmd/cdi-importer/importer.go#L101C1-L102C1

Reply:
still looks redundant...